pkg/scheduler/objects/application.go

```diff
@@ -748,6 +751,7 @@ func (sa *Application) deallocateAsk(ask *Allocation) (*resources.Resource, erro
 
 	delta := ask.GetAllocatedResource()
 	sa.pending = resources.Add(sa.pending, delta)
+	sa.pending.Prune()
```

Comment:
ask or alloc should not have zero types, no need to prune
The delta that gets returned is also dropped by all callers or expected to be nil. Except for 1 tests which does not really need it. Not sure we should keep it.

Reply:
Addressed in latest PR, thanks!